### PR TITLE
18 defence llm evaluation

### PR DESCRIPTION
**Pull request by heatherlogan-scottlogic**

### CODE REVIEW

**Review by gsproston-scottlogic — APPROVED** [review]
Looks great! I've added a couple of comments but they're more out of curiosity than anything else.

File: backend/src/langchain.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  const embeddings = new OpenAIEmbeddings();
+  const vectorStore = await MemoryVectorStore.fromDocuments(
+    docs,
+    embeddings,
+  );
+  // initialise model
+  const model = new ChatOpenAI({
+    modelName: "gpt-4",
+    stream: true,
+  });
```
[the remark this answers is not in the document]
Comment (heatherlogan-scottlogic) [reply backend/src/langchain.js:59]: It takes the env variable

File: backend/src/langchain.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    outputKey: "promptInjectionEval"
+  });
+
+  // create chain to detect malicious prompts
+  const maliciousInputPrompt = PromptTemplate.fromTemplate(maliciousPromptTemplate);
+  const maliciousInputChain = new LLMChain({ 
+    llm: new OpenAI({ model: "gpt-3.5-turbo", temperature: 0 }), 
+    prompt: maliciousInputPrompt,
+    outputKey: "maliciousInputEval"
+  });
```
[the remark this answers is not in the document]
Comment (heatherlogan-scottlogic) [reply backend/src/langchain.js:90]: Yeah we can, although not actually sure if these bigger models will be necessary here. Will have a play around

[the remark this answers is not in the document]
Comment (gsproston-scottlogic) [reply backend/src/langchain.js:90]: ahh yeah that makes sense

**Pull request merged**